python/lsst/drp/tasks/wcsfit.py

```diff
@@ -0,0 +1,1019 @@
+        default=["INSTRUMENT/DEVICE", "EXPOSURE"]
+    )
+    deviceModel = pexConfig.ListField(
+        dtype=str,
+        doc="List of mappings to apply to transform from detector pixels to intermediate frame",
+        default=["BAND/DEVICE/poly"]
+    )
+    exposureModel = pexConfig.ListField(
+        dtype=str,
+        doc="List of mappings to apply to transform from intermediate frame to sky coordinates",
+        default=["EXPOSURE/poly"]
+    )
+    devicePolyOrder = pexConfig.Field(
+        dtype=int,
+        doc="Order of device polynomial model",
+        default=4
+    )
+    exposurePolyOrder = pexConfig.Field(
+        dtype=int,
+        doc="Order of exposure polynomial model",
+        default=6
+    )
+    useProperMotion = pexConfig.Field(
+        dtype=bool,
+        doc="Fit the proper motions of the objects",
+        default=False
+    )
+    verbose = pexConfig.Field(
+        dtype=int,
+        doc="Verbosity level=0, 1, or 2",
+        default=2
+    )
+    requireProperMotion = pexConfig.Field(
+        dtype=bool,
+        doc="Require that reference catalog options have proper motion information",
+        default=False
+    )
+    anyFilterMapsToThis = pexConfig.Field(
+        dtype=str,
+        doc="Use this filter when loading reference objects",
+        default="phot_g_mean"
+    )
+
+    def setDefaults(self):
+        # Use only stars because aperture fluxes of galaxies are biased and
+        # depend on seeing.
+        self.sourceSelector["science"].doUnresolved = True
+        self.sourceSelector["science"].unresolved.name = "extendedness"
+
+        # Use only isolated sources.
+        self.sourceSelector["science"].doIsolated = True
+        self.sourceSelector["science"].isolated.parentName = "parentSourceId"
+        self.sourceSelector["science"].isolated.nChildName = "deblend_nChild"
+        # Do not trust either flux or centroid measurements with flags,
+        # chosen from the usual QA flags for stars.
+        self.sourceSelector["science"].doFlags = True
+        badFlags = ["pixelFlags_edge",
+                    "pixelFlags_saturated",
+                    "pixelFlags_interpolatedCenter",
+                    "pixelFlags_interpolated",
+                    "pixelFlags_crCenter",
+                    "pixelFlags_bad",
+                    "hsmPsfMoments_flag",
+                    f"{self.sourceFluxType}_flag",
+                    ]
+        self.sourceSelector["science"].flags.bad = badFlags
+
+
+class WCSFitTask(pipeBase.PipelineTask):
+    """Calibrate the WCS across multiple visits of the same field.
+    """
+
+    ConfigClass = WCSFitConfig
+    _DefaultName = "wcsfit"
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.makeSubtask("sourceSelector")
+        self.makeSubtask("referenceSelector")
+
+        self.detectors = []
+        self.visits = []
```

Comment:
It is strongly recommended that Tasks be stateless. You hold other instance variables later in this code as well (e.g. `extensionInfo`); can you restructure it to get rid of that state? See the final paragraph of this section:

https://pipelines.lsst.io/v/daily/modules/lsst.pipe.base/creating-a-task.html#the-run-method